pkg/merkle/tree.go

```diff
@@ -0,0 +1,150 @@
+package merkle
+
+import (
+	"errors"
+	"math"
+	"math/bits"
+)
+
+// MerkleTree represents a Merkle tree data structure.
+type MerkleTree struct {
+	tree      [][]byte
```

Comment:
my brain refuses to process anything related to `[][]`. 

If I were writing this, I would typedef the node so at least its `tree []Node`

Even if it is as simple as 
```
type Node struct {
	Hash []byte
}
```